### PR TITLE
WIP dev/translation#13 Changes to Language Switcher

### PR DESCRIPTION
**Pull request by aydun**
Make 'Available Languages' config available in mono-lingual
Make 'Available Languages' select from all installed languages
Change 'languageLimit' setting from storing language string as keys of array to values
Updates to Localization Settings page

See https://lab.civicrm.org/dev/translation/issues/13 for background

### CODE REVIEW

File: CRM/Admin/Form/Setting/Localization.php [hunk 1]
```diff
@@ -63,15 +63,25 @@ public function buildQuickForm() {
     CRM_Utils_System::setTitle(ts('Settings - Localization'));
 
     $warningTitle = json_encode(ts("Warning"));
-    $defaultLocaleOptions = CRM_Admin_Form_Setting_Localization::getDefaultLocaleOptions();
 
-    $domain = new CRM_Core_DAO_Domain();
-    $domain->find(TRUE);
+    $languageLimit = &$this->addElement('advmultiselect', 'languageLimit',
```
Comment (eileenmcnaughton) [inline CRM/Admin/Form/Setting/Localization.php]: @colemanw we killed this didn't we? What is the recommended QF element?

File: CRM/Core/BAO/ConfigSetting.php [hunk 2]
```diff
@@ -141,56 +141,53 @@ public static function applyLocale($settings, $activatedLocales) {
...
+    $requestLocale = CRM_Utils_Request::retrieve('lcMessages', 'String');
+    if (in_array($requestLocale, $languageLimit)) {
+      $chosenLocale = $requestLocale;
 
-      $requestLocale = CRM_Utils_Request::retrieve('lcMessages', 'String');
-      if (in_array($requestLocale, array_keys($languageLimit))) {
-        $chosenLocale = $requestLocale;
+      //CRM-8559, cache navigation do not respect locale if it is changed, so reseting cache.
+      // Ed: This doesn't sound good.
+      CRM_Core_BAO_Cache::deleteGroup('navigation');
```
Comment (eileenmcnaughton) [inline CRM/Core/BAO/ConfigSetting.php]: @totten FYI - may tie into stuff you are working on

**Comment by aydun** [comment]
Hi *@eileenmcnaughton*
*Xavier commented on the gitlab ticket but his use case is different.  It
is ready for review - just needs a reviewer familiar with multi-lang and
translation*

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by mattwire** [comment]
test this please

**Comment by mattwire** [comment]
@mlutfy Could you review this please from a translation / multi-lang perspective?

**Comment by eileenmcnaughton** [comment]
@aydun I think the test fail relates

```
Installing core124055_mijy9 database
not ok 1 - Failure: CRM_ACL_ListTest::testViewAllPermission
  ---
  message: 'Failure in api call for contact create:  in_array() expects parameter 2 to be array, null given'
  severity: fail
  data:
      got: 1
      expected: 0
  ...
not ok 2 - Failure: CRM_ACL_ListTest::testEditAllPermission
  ---
  message: 'Failure in api call for contact create:  Smarty error: unable to read resource: &quot;string:Dear Logged In&quot;'
  severity: fail
  data:
      got: 1
      expected: 0
  ...
not ok 3 - Failure: CRM_ACL_ListTest::testViewEditDeleted
```

**Comment by aydun** [comment]
Hi @eileenmcnaughton, there are lots of errors, but I'm not sure the ones you pointed out relate to the language switcher (they might have done for my permissions PR but that was merged ok)

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
@aydun let's rerun & see

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by mattwire** [comment]
test this please

**Comment by aydun** [comment]
Closing this - replaced by #13240

**Pull request closed without merging**